code/modules/research/xenobiology/crossbreeding/_weapons.dm

```diff
@@ -0,0 +1,112 @@
+/obj/item/kitchen/knife/rainbowknife
+	name = "rainbow knife"
+	desc = "A strange, transparent knife which constantly shifts color. It hums slightly when moved."
+	icon = 'icons/obj/slimecrossing.dmi'
+	icon_state = "rainbowknife"
+	item_state = "rainbowknife"
+	force = 15
+	throwforce = 15
+	damtype = BRUTE
+
+/obj/item/kitchen/knife/rainbowknife/afterattack(atom/O, mob/user, proximity)
+	if(proximity && istype(O, /mob/living))
+		damtype = pick(BRUTE, BURN, TOX, OXY, CLONE)
+	switch(damtype)
+		if(BRUTE)
+			hitsound = 'sound/weapons/bladeslice.ogg'
+			attack_verb = list("slashed","sliced","cut")
+		if(BURN)
+			hitsound = 'sound/weapons/sear.ogg'
+			attack_verb = list("burned","singed","heated")
+		if(TOX)
+			hitsound = 'sound/weapons/pierce.ogg'
+			attack_verb = list("poisoned","dosed","toxified")
+		if(OXY)
+			hitsound = 'sound/effects/space_wind.ogg'
+			attack_verb = list("suffocated","winded","vacuumed")
+		if(CLONE)
+			hitsound = 'sound/items/geiger/ext1.ogg'
+			attack_verb = list("irradiated","mutated","maligned")
+	return ..()
+
+//Adamantine shield - Chilling Adamantine
+/obj/item/twohanded/required/adamantineshield
+	name = "adamantine shield"
+	desc = "A gigantic shield made of solid adamantium."
+	icon = 'icons/obj/slimecrossing.dmi'
+	icon_state = "adamshield"
+	item_state = "adamshield"
+	w_class = WEIGHT_CLASS_HUGE
+	armor = list("melee" = 50, "bullet" = 50, "laser" = 50, "energy" = 0, "bomb" = 30, "bio" = 0, "rad" = 0, "fire" = 80, "acid" = 70)
+	slot_flags = ITEM_SLOT_BACK
+	block_chance = 75
+	throw_range = 1 //How far do you think you're gonna throw a solid crystalline shield...?
+	throw_speed = 2
+	force = 15 //Heavy, but hard to wield.
+	attack_verb = list("bashed","pounded","slammed")
+	item_flags = SLOWS_WHILE_IN_HAND
+
+//Bloodchiller - Chilling Green
+/obj/item/gun/magic/bloodchill
+	name = "blood chiller"
+	desc = "A horrifying weapon made of your own bone and blood vessels. It shoots slowing globules of your own blood. Ech."
+	icon = 'icons/obj/slimecrossing.dmi'
+	icon_state = "bloodgun"
+	item_state = "bloodgun"
+	lefthand_file = 'icons/mob/inhands/weapons/guns_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/guns_righthand.dmi'
+	item_flags = ABSTRACT | NODROP | DROPDEL
+	w_class = WEIGHT_CLASS_HUGE
+	force = 5
+	max_charges = 1 //Recharging costs blood.
+	recharge_rate = 1
+	ammo_type = /obj/item/ammo_casing/magic/bloodchill
+	fire_sound = 'sound/effects/attackblob.ogg'
+
+/obj/item/gun/magic/bloodchill/process()
+	charge_tick++
+	if(charge_tick < recharge_rate || charges >= max_charges)
+		return 0
+	charge_tick = 0
+	var/mob/living/M = loc
+	if(istype(M) && M.blood_volume >= 20)
+		charges++
+		M.blood_volume -= 20
+	if(charges == 1)
+		recharge_newshot()
+	return 1
+
+/obj/item/ammo_casing/magic/bloodchill
+	projectile_type = /obj/item/projectile/magic/bloodchill
+
+/obj/item/projectile/magic/bloodchill
+	name = "blood ball"
+	icon_state = "pulse0_bl"
+	damage = 0
```

Comment:
I feel like this should deal SOME damage- people are paying in their own blood for this, and it's a bloody (get it?) Megaman arm cannon. I fear that not many people will want to "switch-hit".

Reply:
Maybe stetchkin damage? This doesn't have anywhere near the rate of fire of a stetchkin, so it should be okay balance-wise (in comparison to the stetchkin, at least).

Reply:
Ooh, what if you could switch it between "lethal" and "nonlethal" modes, with the "lethal" mode being the same as the "nonlethal" mode, but costing more charges/blood and dealing damage in addition to the slowdown effect?

Reply:
Chilling extracts were/are supposed to be the defensive themed counterparts to burning, hence the lack of damage. If you guys think I should add damage, I could totally do it - Potentially by making it do minor cold damage over time with the status effect.

Reply:
Opted to implement a 50% chance to get 2 burn damage each second as part of the status effect.